java-checks-test-sources/default/src/main/java/checks/ConstantsShouldBeStaticFinalCheck.java

```diff
@@ -116,3 +136,16 @@ public void printCreation() {
   }
 }
 
+class FieldAssignments {
+  private final int[] finalFoo = new int[42]; // Compliant
+  private final Object finalObject = finalFoo; // Compliant - finalFoo is not static
+
+  private static int[] staticFoo = new int[42]; // Compliant
+
+  private final Object staticObject = staticFoo; // Compliant - staticFoo is not final
+
+  private static final int[] foo = new int[42];
+  private final Object object = foo; // Noncompliant
+
```

Comment:
Additional test case for the array index fix (see below)

```suggestion
private final int bar = foo[2]; // Compliant

```